packages/common/core-interfaces/src/exposedUtilityTypes.ts

```diff
@@ -0,0 +1,549 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @rushstack/no-new-null */
+
+import type {
+	SerializationErrorPerNonPublicProperties,
+	SerializationErrorPerUndefinedArrayElement,
+} from "./jsonSerializationErrors.js";
+import type { JsonTypeWith, NonNullJsonObjectWith } from "./jsonType.js";
+
+/**
+ * Collection of utility types that are not intended to be used/imported
+ * directly outside of this package.
+ *
+ * @beta
+ * @system
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace InternalUtilityTypes {
```

Comment:
Nit: Consider renaming to 'InternalTypes' for consistency with tree DDS?

```ts
import * as InternalTypes from "./internalTypes.js";
export {
	/**
	 * Contains types used by the API, but which serve mechanical purposes and do not represent semantic concepts.
	 * They are used internally to implement API aspects, but are not intended for use by external consumers.
	 */
	InternalTypes,
};
```

Reply:
I am expecting that we are going to end up with multiple namespaces in some packages for multiple internal things and the only thing that needs to be consistent (my opinion) is `Internal` prefix on the namespace. None of these should be imported from outside of the package (darn api-extractor); so, I think name that makes sense for internal description is good AND we should be able to arbitrarily change these names without it being considered a breaking change for customers.

BTW tree is also investigating use of explicit namespaces similar to what I used here. I tried the tree pattern show above, but api-extractor is unhappy about `{@link }` references when you rename like that.